database/migrations/2019_07_12_045144_create_plans_table.php

```diff
@@ -0,0 +1,92 @@
+                $table->foreign('plan_id', 'FK_plan_days')->references('id')->on(config('database.connections.dbp_users.database') . '.plans')->onDelete('cascade')->onUpdate('cascade');
+                $table->timestamp('created_at')->default(DB::raw('CURRENT_TIMESTAMP'));
+                $table->timestamp('updated_at')->default(DB::raw('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP'));
+                $table->integer('order_column');
+            });
+        }
+
+        if (!Schema::connection('dbp_users')->hasTable('plan_playlist_items')) {
+            Schema::connection('dbp_users')->create('plan_playlist_items', function (Blueprint $table) {
+                $table->bigIncrements('id');
+                $table->bigInteger('day_id')->unsigned()->nullable();
+                $table->foreign('day_id', 'FK_plan_days_items')->references('id')->on(config('database.connections.dbp_users.database') . '.plan_days')->onDelete('cascade')->onUpdate('cascade');
+                $table->bigInteger('playlist_id')->unsigned()->nullable();
+                $table->foreign('playlist_id', 'FK_playlists_items')->references('id')->on(config('database.connections.dbp_users.database') . '.user_playlists')->onDelete('cascade')->onUpdate('cascade');
+                $table->string('fileset_id', 16);
+                $table->foreign('fileset_id', 'FK_bible_filesets_plan_days_items')->references('id')->on(config('database.connections.dbp.database') . '.bible_filesets')->onUpdate('cascade')->onDelete('cascade');
+                $table->char('book_id', 3);
+                $table->foreign('book_id', 'FK_books_plan_days_items')->references('id')->on(config('database.connections.dbp.database') . '.books');
+                $table->integer('chapter_start')->unsigned();
+                $table->integer('chapter_end')->unsigned()->nullable();
+                $table->integer('verse_start')->unsigned();
+                $table->integer('verse_end')->unsigned()->nullable();
+                $table->integer('duration');
+                $table->timestamp('created_at')->default(DB::raw('CURRENT_TIMESTAMP'));
+                $table->timestamp('updated_at')->default(DB::raw('CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP'));
+                $table->integer('order_column');
+            });
+        }
+
+        if (!Schema::connection('dbp_users')->hasTable('user_plans')) {
+            Schema::connection('dbp_users')->create('user_plans', function (Blueprint $table) {
+                $table->index(['user_id', 'plan_id']);
+                $table->integer('user_id')->unsigned();
+                $table->foreign('user_id', 'FK_users_user_plans')->references('id')->on(config('database.connections.dbp_users.database') . '.users')->onUpdate('cascade');
+                $table->bigInteger('plan_id')->unsigned();
+                $table->date('start_date')->nullable();
+                $table->integer('percentage_completed');
+                $table->text('days_completed');
```

Comment:
how will user's progress on a plan be stored?  if in this field, i'm concerned that it is a text type, also with below items_completed.  is the intent to put a list of play_day_id's in here, and similar for below items_completed?  if so, seems better to put in separate tables, eg a table for completed_plan_items with at least {user_id (FK), plan_playlist_item_id (FK)}, with entries made upon completing each item.  and a similar table for completed_plan_days.  my impression of your intent for these _completed fields may be incorrect - please describe.

Reply:
Done Jon, now are created the tables `plan_days_completed` and `playlist_items_completed`